components/raftstore/src/store/peer_storage.rs

```diff
@@ -784,24 +995,35 @@ where
         let cache_low = self.cache.first_index().unwrap_or(u64::MAX);
         if high <= cache_low {
             self.cache.miss.update(|m| m + 1);
-            self.engines.raft.fetch_entries_to(
-                region_id,
-                low,
-                high,
-                Some(max_size as usize),
-                &mut ents,
-            )?;
-            return Ok(ents);
+            return if context.can_async() {
+                self.async_fetch(region_id, low, high, max_size, context, &mut ents)?;
```

Comment:
I'm afraid it may cause serious regression even under normal workload. It may make the slow node always lagging behind as it always takes more time to fetch logs.

Reply:
test in under 8c with sysbench-insert(400, 800 concurrency), sysbench-write-only(400, 800 concurrency) which is the case for testing async-io

Under normal workload, it can hardly trigger async fetch, See
<img width="1021" alt="截屏2022-02-07 下午12 33 10" src="https://user-images.githubusercontent.com/13497871/152725214-84a047e3-64dd-42b1-99ff-6de5445fee1f.png">
<img width="1022" alt="截屏2022-02-07 下午12 33 26" src="https://user-images.githubusercontent.com/13497871/152725284-f2e854bb-f961-42e9-8968-c99cec23676a.png">

Reply:
It seems async fetches are always triggered when the workload is finished.

Reply:
because there is a restart